library/psa_crypto.c

```diff
@@ -5772,7 +5855,8 @@ psa_status_t psa_key_derivation_input_key(
 /* Key agreement */
 /****************************************************************/
 
-#if defined(MBEDTLS_ECDH_C)
+#if defined(MBEDTLS_PSA_BUILTIN_ALG_ECDH) && \
+    defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR)
```

Comment:
I'm interpreting (but this interpretation may be wrong) MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR as: I want the builtin code to support for ECC key pair management (import, generate, copy, derive ?). But this support could rather rely on an accelerator that is not ECDH capable and we would need this function even if MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR is not defined. In that case it should be only "#if defined(MBEDTLS_PSA_BUILTIN_ALG_ECDH)" here.

Reply:
Your logic makes sense here. I am updating to match the requested change.